### DISCUSSION TITLE
Squelch in AM detector

### DISCUSSION THREAD

**Discussion by ah1102** [DSheirer/sdrtrunk#1441] [Ideas]
Please can you add a squelch to the AM detector? In NBFM mode, the squelch works great! Just copy the same code for AM detector mode.

In NBFM, it is now possible to monitor amateur radio repeaters and radio amateurs operating in this non-digital mode. I would like to be able to monitor aircraft traffic. Without noise suppression, this is not possible.

![Снимок экрана 2023-02-11 в 16 11 04](https://user-images.githubusercontent.com/58341369/218259834-8390aec2-0766-437d-af26-b48b0d026bc2.png)

**Comment by ah1102** [comment]
In AM mode, a tab with settings for the demodulator is prepared, but it is not active.

Is it possible to just port the NBFM code of the squelch to have the squelch in AM mode?

![Снимок экрана 2023-02-19 в 16 14 59](https://user-images.githubusercontent.com/58341369/219950416-71896bf5-b458-4977-b0fe-e2b5919c2f4c.png)

**Reply by hans109h** [reply]
ah1102, if it was as simple as using FM squelch for AM the world would be a very different place.  Detection and suppression of noise/silence/transmission of the two differs greatly.  AM squelch isn't impossible, but it is as easy as what you describe so is very unlikely to be implemented with out a lot of addition coding.

**Comment by DSheirer** [comment]
@ah102 support for AM Squelch is a planned feature for 0.6.0
